trio/_subprocess.py

```diff
@@ -835,10 +885,203 @@ async def run_process(
+                  ``returncode`` attribute of the returned object if you pass
+                  ``check=False``, so that errors don't pass silently.
+
+              deliver_cancel (async function or None): If `run_process` is cancelled,
+                  then it needs to kill the child process. There are multiple ways to
+                  do this, so we let you customize it.
+
+                  If you pass None (the default), then the behavior depends on the
+                  platform:
+
+                  - On Windows, Trio calls ``TerminateProcess``, which should kill the
+                    process immediately.
+
+                  - On Unix-likes, the default behavior is to send a ``SIGTERM``, wait
+                    5 seconds, and send a ``SIGKILL``.
+
+                  Alternatively, you can customize this behavior by passing in an
+                  arbitrary async function, which will be called with the `Process`
+                  object as an argument. For example, the default Unix behavior could
+                  be implemented like this::
+
+                     async def my_deliver_cancel(process):
+                         process.send_signal(signal.SIGTERM)
+                         await trio.sleep(5)
+                         process.send_signal(signal.SIGKILL)
+
+                  When the process actually exits, the ``deliver_cancel`` function
+                  will automatically be cancelled – so if the process exits after
+                  ``SIGTERM``, then we'll never reach the ``SIGKILL``.
+
+                  In any case, `run_process` will always wait for the child process to
+                  exit before raising `Cancelled`.
+
+              **options: :func:`run_process` also accepts any :ref:`general subprocess
+                  options <subprocess-options>` and passes them on to the
+                  :class:`~trio.Process` constructor. This includes the
+                  ``stdout`` and ``stderr`` options, which provide additional
+                  redirection possibilities such as ``stderr=subprocess.STDOUT``,
+                  ``stdout=subprocess.DEVNULL``, or file descriptors.
+
+            Returns:
+
+              When called normally – a `subprocess.CompletedProcess` instance
+              describing the return code and outputs.
+
+              When called via `Nursery.start` – a `trio.Process` instance.
+
+            Raises:
+              UnicodeError: if ``stdin`` is specified as a Unicode string, rather
+                  than bytes
+              ValueError: if multiple redirections are specified for the same
+                  stream, e.g., both ``capture_stdout=True`` and
+                  ``stdout=subprocess.DEVNULL``
+              subprocess.CalledProcessError: if ``check=False`` is not passed
+                  and the process exits with a nonzero exit status
+              OSError: if an error is encountered starting or communicating with
+                  the process
```

Comment:
Again note about docstring format

Reply:
same thing - copypasted from `run_process` earlier in the file. (and yes duplicate docstrings is incredibly ugly and bad... but I don't know how to improve @TeamSpen210's magic to avoid needing to and have it work both on runtime, in static analysis, and on all platforms.)

Reply:
TBH sphinx renders this fine so whatever.